include/gsl/gsl-lite.hpp

```diff
@@ -612,7 +612,12 @@
 #endif
 
 #if gsl_HAVE( TYPE_TRAITS )
-# define gsl_REQUIRES_T_(VA) , typename = typename std::enable_if< ( VA ), ::gsl::detail::enabler >::type
+# if gsl_BETWEEN( gsl_COMPILER_MSVC_VERSION, 1, 140 )
+// VS 2013 and earlier seem to have trouble with SFINAE for default non-type arguments
+#  define gsl_REQUIRES_T_(VA) , typename = typename std::enable_if< ( VA ), ::gsl::detail::enabler >::type
+# else
+#  define gsl_REQUIRES_T_(VA) , typename std::enable_if< ( VA ), int >::type = 0
+# endif
```

Comment:
What is the reason for this change? I understand that we want to use `typename std::enable_if< X, int >::type = 0` rather than `typename = typename std::enable_if< X >::type` in *function templates*, but we already have `gsl_REQUIRES_A_()` for that purpose.

If the latter form is not required anywhere, perhaps we should remove `gsl_REQUIRES_T_()` altogether?

Reply:
Ah great, you awoke me :) I have generalized the note about *function templates* to include class templates :)

Thus there is no reason for this change left.